Jellyfin.Networking/HappyEyeballs/HttpClientExtension.cs

```diff
@@ -0,0 +1,122 @@
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+using System.IO;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Jellyfin.Networking.HappyEyeballs
+{
+    /// <summary>
+    /// Defines the <see cref="HttpClientExtension"/> class.
+    ///
+    /// Implementation taken from https://github.com/ppy/osu-framework/pull/4191 .
+    /// </summary>
+    public static class HttpClientExtension
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether the client should use IPv6.
+        /// </summary>
+        public static bool UseIPv6 { get; set; } = true;
+
+        /// <summary>
+        /// Implements the httpclient callback method.
+        /// </summary>
+        /// <param name="context">The <see cref="SocketsHttpConnectionContext"/> instance.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> instance.</param>
+        /// <returns>The http steam.</returns>
+        public static async ValueTask<Stream> OnConnect(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
+        {
+            if (!UseIPv6)
+            {
+                return await AttemptConnection(AddressFamily.InterNetwork, context, cancellationToken).ConfigureAwait(false);
+            }
+
+            using var cancelIPv6 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            var tryConnectAsyncIPv6 = AttemptConnection(AddressFamily.InterNetworkV6, context, cancelIPv6.Token);
+
+            //This connect task uses GetAwaiter().GetResult() as the appropriate task has already been completed.
+            //This results in improved exception handling.
+            //See https://github.com/dotnet/corefx/pull/29792/files#r189415885 for more details.
+            if (await Task.WhenAny(tryConnectAsyncIPv6, Task.Delay(200, cancelIPv6.Token)).ConfigureAwait(false) == tryConnectAsyncIPv6 && tryConnectAsyncIPv6.IsCompletedSuccessfully)
+            {
+                cancelIPv6.Cancel();
+                return tryConnectAsyncIPv6.GetAwaiter().GetResult();
+            }
+
+            using var cancelIPv4 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            var tryConnectAsyncIPv4 = AttemptConnection(AddressFamily.InterNetwork, context, cancelIPv4.Token);
+
+            //Both connect tasks use GetAwaiter().GetResult() as the appropriate task has already been completed.
+            //This results in improved exception handling.
+            if (await Task.WhenAny(tryConnectAsyncIPv6, tryConnectAsyncIPv4).ConfigureAwait(false) == tryConnectAsyncIPv6)
```

Comment:
One comment should suffice
```suggestion
            if (await Task.WhenAny(tryConnectAsyncIPv6, tryConnectAsyncIPv4).ConfigureAwait(false) == tryConnectAsyncIPv6)
```